Storage/FirebaseStorageUI/SDWebImageFIRStorageLoader.m

```diff
@@ -0,0 +1,107 @@
+  FIRStorageReference *storageRef = url.sd_storageReference;
+  if (!storageRef) {
+    if (completedBlock) {
+      NSError *error = [NSError errorWithDomain:SDWebImageErrorDomain code:SDWebImageErrorInvalidURL userInfo:@{NSLocalizedDescriptionKey : @"The provided image url must have an associated FIRStorageReference."}];
+      completedBlock(nil, nil, error, YES);
+    }
+  }
+  
+  UInt64 size;
+  if (context[SDWebImageContextFIRStorageMaxImageSize]) {
+    size = [context[SDWebImageContextFIRStorageMaxImageSize] unsignedLongLongValue];
+  } else {
+    size = self.defaultMaxImageSize;
+  }
+  // Download the image from Firebase Storage
+  
+  // TODO: Support progressive image loading using the `GTMSessionFetcher.downloadedData` with `SDImageLoaderDecodeProgressiveImageData`
+  FIRStorageDownloadTask * download = [storageRef dataWithMaxSize:size
+                                                       completion:^(NSData * _Nullable data, NSError * _Nullable error) {
+                                                         if (error) {
+                                                           dispatch_main_async_safe(^{
+                                                             if (completedBlock) {
+                                                               completedBlock(nil, nil, error, YES);
+                                                             }
+                                                           });
+                                                           return;
+                                                         }
+                                                         // Decode the image with data
+                                                         dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_HIGH, 0), ^{
+                                                           UIImage *image = SDImageLoaderDecodeImageData(data, url, options, context);
+                                                           dispatch_main_async_safe(^{
+                                                             if (completedBlock) {
+                                                               completedBlock(image, data, nil, YES);
+                                                             }
+                                                           });
+                                                         });
+                                                       }];
+  // Observe the progress changes
+  [download observeStatus:FIRStorageTaskStatusProgress handler:^(FIRStorageTaskSnapshot * _Nonnull snapshot) {
+    NSProgress *progress = snapshot.progress;
+    // completedUnitCount == totalBytesWritten;
+    // totalUnitCount == totalBytesExpectedToWrite;
```

Comment:
nit: Remove comments if they're unused

Reply:
Good. This is previously written here because I want to support the progressive image loading. (Which is formally support by SDWebImage for HTTP/HTTPS requests). But then I found seems there are no Public API for FirebaseStorage, to get the partial image data. I'll delete this.

Maybe I need to submit a feature request to upstream FirebaseStorage repo?

Reply:
That's a good feature request idea 👍

Maybe replace the comments here with a TODO?